src/flash-source-buffer.js

```diff
@@ -182,11 +182,18 @@ export default class FlashSourceBuffer extends videojs.EventTarget {
       removeCuesFromTrack(0, Infinity, this.inbandTextTrack_);
     });
 
+    this.mediaSource_.player_.tech_.on('hls-reset', this.resetHls);
```

Comment:
For some reason when `this.resetHls` is called after hearing this event, `this` within `resetHls` is not being bound correctly.

if you define the function above instead of using the es6 class syntax it works
```
this.resetHls = () => {
  this.transmuxer_.postMessage({action: 'resetCaptions'});
};

this.mediaSource_.player.tech_.on('hls-reset', .......
```